### PR TITLE
Dev phpunit

### PR DESCRIPTION
**Pull request by tuanhai2712**
## Trello ticket URL
-  Write PHPUnit test

## Technical changes overview
- No

## SQL update
- No

## Other issues
- No

### CODE REVIEW

File: config/database.php [hunk 1]
```diff
@@ -35,7 +35,7 @@
 
         'sqlite' => [
```
Comment (ngon12) [inline config/database.php]: Please add gitignore for this file. This is config file so you don't need push.

File: phpunit.xml [hunk 2]
```diff
@@ -27,5 +27,7 @@
         <env name="CACHE_DRIVER" value="array"/>
```
Comment (ngon12) [inline phpunit.xml]: This is config file so don;t need to push

File: app/Http/Controllers/MemberProjectController.php [hunk 3]
```diff
@@ -2,38 +2,33 @@
...
 namespace App\Http\Controllers;
 
+use Illuminate\Http\Response;
 use Illuminate\Http\Request;
-use App\Member;
-use App\Project;
+use App\MemberProject;
 
-class MemberProjectController extends Controller
+class MemberProjectController extends request
```
Comment (4c0n) [inline app/Http/Controllers/MemberProjectController.php]: Why does the controller extend a class named `request`?

File: app/Http/Controllers/MemberProjectController.php [hunk 4]
```diff
@@ -2,38 +2,33 @@
 
 namespace App\Http\Controllers;
 
+use Illuminate\Http\Response;
```
Comment (4c0n) [inline app/Http/Controllers/MemberProjectController.php]: I don't see the Response class being used in this class, why are you importing it?

File: app/Http/Controllers/ProjectController.php [hunk 5]
```diff
@@ -3,63 +3,44 @@
 namespace App\Http\Controllers;
 
 use Illuminate\Http\Request;
-use App\Http\Requests\ProjectValidatorRequest;
+use Illuminate\Http\Response;
```
Comment (4c0n) [inline app/Http/Controllers/ProjectController.php]: Why are you importing the Response class and not using it?

File: app/Http/Middleware/VerifyCsrfToken.php [hunk 6]
```diff
@@ -12,7 +12,8 @@ class VerifyCsrfToken extends Middleware
      * @var array
      */
     protected $except = [
-        'createproject',
-        'createmember', 'editmember'
+        'member/store','member/update/{id}', 'member/update', 'member/destroy',
```
Comment (4c0n) [inline app/Http/Middleware/VerifyCsrfToken.php]: With this you are disabling the Cross Site Request Forgery protection. https://laravel.com/docs/5.5/csrf
So basically this is not secure anymore.

File: app/MemberProject.php [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        ];
+        $query = MemberProject::where($array)->count();
+        if ($query === 0) {
+            $newMemberProject->save();
+            return $newMemberProject;
+        }
+        return response()->json(['Failed' => 'Already exists in the database'], 422);
+    }
+
+    public function updateMemberProject($request)
```
Comment (4c0n) [inline app/MemberProject.php]: Please add a type hint or doc block so we can know what kind of object `$request` is.

File: app/MemberProject.php [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+class MemberProject extends Model
+{
+    protected $fillable = [
+        'member_id', 'project_id'
+    ];
+
+    public $table = "member_project";
+
+    public function store($request)
```
Comment (4c0n) [inline app/MemberProject.php]: Please add a type hint or doc block so we can know what kind of object $request is.

File: app/MemberProject.php [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+            $updateMemberProject = MemberProject::find($id);
+            $updateMemberProject->member_id = $data['member_id'];
+            $updateMemberProject->project_id = $data['project_id'];
+            $updateMemberProject->save();
+            return $updateMemberProject;
+        }
+        return response()->json(['Failed' => 'Already exists in the database'], 422);
+    }
+
+    public function destroyMemberProject($request)
```
Comment (4c0n) [inline app/MemberProject.php]: Please add a type hint or doc block so we can know what kind of object $request is.

File: app/Project.php [hunk 10]
```diff
@@ -18,4 +18,52 @@ public function member()
     }
 
     public $table = "project";
+
+    public function data($request)
```
Comment (4c0n) [inline app/Project.php]: Please add a type hint or doc block so we can know what kind of object $request is.

File: app/Project.php [hunk 11]
```diff
@@ -18,4 +18,52 @@ public function member()
...
+        return $profileProject;
+    }
+
+    public function index()
+    {
+        $index = Project::all()->toArray();
+        return $index;
+    }
+
+    public function store($request)
```
Comment (4c0n) [inline app/Project.php]: Please add a type hint or doc block so we can know what kind of object $request is.

File: app/Project.php [hunk 12]
```diff
@@ -18,4 +18,52 @@ public function member()
...
+        $profileProject->project_deadline = $data['project_deadline'];
+        $profileProject->project_type = $data['project_type'];
+        $profileProject->project_status = $data['project_status'];
+        $profileProject->save();
+        return $profileProject;
+    }
+
+    public function index()
+    {
+        $index = Project::all()->toArray();
```
Comment (4c0n) [inline app/Project.php]: Why do we need this method? Also you should not return an array, but use the Collections as that is much more flexible especially when it comes to memory usage and pagination.

File: app/Project.php [hunk 13]
```diff
@@ -18,4 +18,52 @@ public function member()
...
+        return $index;
+    }
+
+    public function store($request)
+    {
+        $store = $this->data($request);
+        return $store;
+    }
+
+    public function viewUpdate($id)
```
Comment (4c0n) [inline app/Project.php]: Same don't return an array.

File: app/Project.php [hunk 14]
```diff
@@ -18,4 +18,52 @@ public function member()
...
+
+    public function updateProject($request)
+    {
+        $update = $this->data($request);
+        return $update;
+    }
+
+    public function destroyProject($id)
+    {
+        $destroy = Project::find($id)->delete();
```
Comment (4c0n) [inline app/Project.php]: What happens when there is no project with this id?

File: app/Project.php [hunk 15]
```diff
@@ -18,4 +18,52 @@ public function member()
...
+        return $store;
+    }
+
+    public function viewUpdate($id)
+    {
+        $dataProject = Project::find($id)->toArray();
+        return $dataProject;
+    }
+
+    public function updateProject($request)
```
Comment (4c0n) [inline app/Project.php]: Please add a type hint or docblock to explain what type of object $request is

File: routes/web.php [hunk 16]
```diff
@@ -15,15 +15,24 @@
...
-Route::get('listmember', 'MemberController@listMember');
-Route::get('createmember', 'MemberController@actionGetCreateMember');
-Route::post('createmember', 'MemberController@createMember');
-Route::get('editmember/{id}', 'MemberController@actionGetEditMember')->name('editmember');
-Route::post('editmember', 'MemberController@editMember');
-Route::get('deletemember/{id}', 'MemberController@deleteMember')->name('deletemember');
+Route::group(['prefix' => 'member'], function () {
+    Route::get('list', 'MemberController@index');
+    Route::post('store', 'MemberController@store');
+    Route::get('update/{id}', 'MemberController@viewUpdate')->name('update');
```
Comment (4c0n) [inline routes/web.php]: What is the purpose of this route?

File: tests/Feature/MemberProjectControllerTest.php [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+    use DatabaseMigrations;
+    use WithoutMiddleware;
+
+    public function testStoreSuccess()
+    {
+        $array = [
+            'member_id' => 1,
+            'project_id' => 2
+        ];
+        $response = $this->json('POST', 'memberproject/store', $array);
```
Comment (4c0n) [inline tests/Feature/MemberProjectControllerTest.php]: You need to add some assertions for the response object. Now you are not verifying that the correct JSON response is returned.

File: tests/Feature/MemberProjectControllerTest.php [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+
+    public function testStoreFailedDueAlreadyExistsInTheDatabase()
+    {
+        $create = Factory(MemberProject::class)->create();
+        $array = [
+            'id' => $create['id'],
+            'member_id' => $create['member_id'],
+            'project_id' => $create['project_id']
+        ];
+        $assertDatabaseHas = $this->assertDatabaseHas('member_project', $array);
```
Comment (4c0n) [inline tests/Feature/MemberProjectControllerTest.php]: This assertion is testing you test code, so you don't need it.

File: tests/Feature/MemberProjectControllerTest.php [hunk 19]
```diff
@@ -0,0 +1,100 @@
...
+        $response = $this->json('POST', 'memberproject/store', $array);
```
Comment (4c0n) [inline tests/Feature/MemberProjectControllerTest.php]: You should verify that your JSON response is correct.

File: tests/Feature/MemberProjectControllerTest.php [hunk 20]
```diff
@@ -0,0 +1,100 @@
...
+
+    public function testUpdateSuccess()
+    {
+        $create = Factory(MemberProject::class)->create();
+        $array = [
+            'id' => $create['id'],
+            'member_id' => $create['member_id'],
+            'project_id' => $create['project_id'],
+        ];
+        $assertDatabaseHas = $this->assertDatabaseHas('member_project', $array);
```
Comment (4c0n) [inline tests/Feature/MemberProjectControllerTest.php]: This assertion is testing your test code, so you don't need it.

File: tests/Feature/MemberProjectControllerTest.php [hunk 21]
```diff
@@ -0,0 +1,100 @@
...
+        $update = [
+            'id' => $create['id'],
+            'member_id' => 2,
+            'project_id' => 3
+        ];
+        $response = $this->json('PUT', 'memberproject/update', $update);
```
Comment (4c0n) [inline tests/Feature/MemberProjectControllerTest.php]: You should verify the JSON response: `$response->assertJson()` https://laravel.com/docs/5.5/http-tests#testing-json-apis

File: tests/Feature/MemberProjectControllerTest.php [hunk 22]
```diff
@@ -0,0 +1,100 @@
...
+
+    public function testUpdateFailedDueAlreadyExistsInTheDatabase()
+    {
+        $create = Factory(MemberProject::class)->create();
+        $array = [
+            'id' => $create['id'],
+            'member_id' => $create['member_id'],
+            'project_id' => $create['project_id']
+        ];
+        $assertDatabaseHas = $this->assertDatabaseHas('member_project', $array);
```
Comment (4c0n) [inline tests/Feature/MemberProjectControllerTest.php]: same

File: tests/Feature/MemberProjectControllerTest.php [hunk 23]
```diff
@@ -0,0 +1,100 @@
...
+        $update = [
+            'id' => $create['id'],
+            'member_id' => $create['member_id'],
+            'project_id' => $create['project_id']
+        ];
+        $response = $this->json('PUT', 'memberproject/update', $update);
```
Comment (4c0n) [inline tests/Feature/MemberProjectControllerTest.php]: same

File: tests/Feature/MemberProjectControllerTest.php [hunk 24]
```diff
@@ -0,0 +1,100 @@
...
+        $this->assertEquals(422, $response->status());
+    }
+
+    public function testDestroySuccess()
+    {
+        $create = Factory(MemberProject::class)->create();
+        $array = [
+            'id' => $create['id']
+        ];
+        $assertDatabaseHas = $this->assertDatabaseHas('member_project', $array);
```
Comment (4c0n) [inline tests/Feature/MemberProjectControllerTest.php]: same

File: tests/Feature/MemberProjectControllerTest.php [hunk 25]
```diff
@@ -0,0 +1,100 @@
...
+        $response = $this->json('DELETE', 'memberproject/destroy', $array);
```
Comment (4c0n) [inline tests/Feature/MemberProjectControllerTest.php]: same

File: tests/Feature/MemberControllerTest.php [hunk 26]
```diff
@@ -0,0 +1,870 @@
+<?php
+
+namespace Tests\Feature;
+
+use Lang;
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: Don't import classes you don't use.

File: tests/Feature/MemberControllerTest.php [hunk 27]
```diff
@@ -0,0 +1,870 @@
...
+use Faker\Factory;
+use App\Member;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Http\Response;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Illuminate\Foundation\Testing\WithoutMiddleware;
+
+class MemberControllerTest extends TestCase
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: Please do not test test code and add assertions to verify your JSON responses.

File: tests/Feature/MemberControllerTest.php [hunk 28]
```diff
@@ -0,0 +1,870 @@
...
+        ]);
+    }
+
+    public function testStoreSuccess()
+    {
+
+        $stub = public_path().'/avatar/test.jpg';
+        $name = str_random(8).'.jpg';
+        $path = public_path().'/avatar/test/'.$name;
+        copy($stub, $path);
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: I think this file should be removed at some point

File: tests/Feature/ProjectControllerTest.php [hunk 29]
```diff
@@ -0,0 +1,457 @@
...
+use Tests\TestCase;
+use Faker\Factory;
+use App\Project;
+use Illuminate\Http\Response;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Illuminate\Foundation\Testing\WithoutMiddleware;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+
+class ProjectControllerTest extends TestCase
```
Comment (4c0n) [inline tests/Feature/ProjectControllerTest.php]: Same feedback as the previous tests.

File: app/Project.php (hunk 12 above)
Comment (tuanhai2712) [reply app/Project.php]: This method is used to retrieve an existing project list
I used the collection according to your instructions

File: routes/web.php (hunk 16 above)
Comment (tuanhai2712) [reply routes/web.php]: When a user submits a request, this route will direct the user's request to the corresponding controller for processing

File: config/database.php (hunk 1 above)
Comment (4c0n) [reply config/database.php]: Actually this file is part of the config files that need to be committed. However actually the settings should be overridden in phpunit.xml or the .env file. But in certain cases it is okay to edit this file directly.

File: tests/Feature/MemberControllerTest.php [hunk 30]
```diff
@@ -0,0 +1,940 @@
...
+        $array = [
+            'member_name' => 'Nguyen Tuan Hai',
+            'member_info' => 'Developer',
+            'member_phone' => '0947271292',
+            'member_dateofbirth' => '1992-12-27',
+            'member_avatar' => $file,
+            'member_position' => 'senior',
+            'member_gender' => 'male'
+        ];
+        $response = $this->json('POST', 'member/store', $array);
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: You are not verifying the JSON response in this test case. Please verify that the correct JSON response is returned.

File: tests/Feature/MemberControllerTest.php [hunk 31]
```diff
@@ -0,0 +1,940 @@
...
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: You are not verifying the JSON response in this test case. Please verify that the correct JSON response is returned.

File: tests/Feature/MemberControllerTest.php [hunk 32]
```diff
@@ -0,0 +1,940 @@
...
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: You are not verifying the JSON response in this test case. Please verify that the correct JSON response is returned.

File: tests/Feature/MemberControllerTest.php [hunk 33]
```diff
@@ -0,0 +1,940 @@
...
+        $editMember = [
+            'member_name' => $array['member_name'],
+            'member_info' => $array['member_info'],
+            'member_phone' => $array['member_phone'],
+            'member_dateofbirth' => $array['member_dateofbirth'],
+            'member_avatar' => $file,
+            'member_position' => $array['member_position'],
+            'member_gender' => $array['member_gender']
+        ];
+        $response = $this->json('PUT', 'member/update', $editMember);
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: You are not verifying the JSON response in this test case. Please verify that the correct JSON response is returned.

File: tests/Feature/MemberControllerTest.php [hunk 34]
```diff
@@ -0,0 +1,940 @@
...
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: You are not verifying the JSON response in this test case. Please verify that the correct JSON response is returned.

File: tests/Feature/ProjectControllerTest.php [hunk 35]
```diff
@@ -0,0 +1,584 @@
...
+            'project_info' => $array['project_info'],
+            'project_deadline' => $array['project_deadline'],
+            'project_type' => $array['project_type'],
+            'project_status' => $array['project_status']
+        ];
+        $response = $this->json('PUT', 'project/update', $editProject);
+        $response
+            ->assertStatus(200)
+            ->assertJson($editProject);
+        $this->assertEquals(200, $response->status());
```
Comment (4c0n) [inline tests/Feature/ProjectControllerTest.php]: This assertion has become redundant

File: tests/Feature/ProjectControllerTest.php [hunk 36]
```diff
@@ -0,0 +1,584 @@
...
+            'project_info' => 'The first project',
+            'project_deadline' => '2018-12-12',
+            'project_type' => 'lab',
+            'project_status' => 'doing'
+        ];
+        $response = $this->json('POST', 'project/store', $array);
+        $response
+            ->assertStatus(200)
+            ->assertJson($array);
+        $this->assertEquals(200, $response->status());
```
Comment (4c0n) [inline tests/Feature/ProjectControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberProjectControllerTest.php [hunk 37]
```diff
@@ -0,0 +1,121 @@
...
+    {
+        $array = Factory(MemberProject::class)->create()->toArray();
+        $this->call('DELETE', 'memberproject/destroy', $array);
+        $response = $this->call('DELETE', 'memberproject/destroy', $array);
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+            'Failed' => 'ID does not exist in the database',
+            ]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberProjectControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberProjectControllerTest.php [hunk 38]
```diff
@@ -0,0 +1,121 @@
...
+    public function testDestroySuccess()
+    {
+        $array = Factory(MemberProject::class)->create()->toArray();
+        $response = $this->json('DELETE', 'memberproject/destroy', $array);
+        $response
+            ->assertStatus(200)
+            ->assertJson([
+            'Success' => 'Delete Success',
+            ]);
+        $this->assertEquals(200, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberProjectControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberProjectControllerTest.php [hunk 39]
```diff
@@ -0,0 +1,121 @@
...
+            'member_id' => $create['member_id'],
+            'project_id' => $create['project_id']
+        ];
+        $response = $this->json('PUT', 'memberproject/update', $update);
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+            'Failed' => 'Already exists in the database',
+            ]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberProjectControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberProjectControllerTest.php [hunk 40]
```diff
@@ -0,0 +1,121 @@
...
+        $update = [
+            'id' => $create['id'],
+            'member_id' => 2,
+            'project_id' => 3
+        ];
+        $response = $this->json('PUT', 'memberproject/update', $update);
+        $response
+            ->assertStatus(200)
+            ->assertJson($update);
+        $this->assertEquals(200, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberProjectControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberProjectControllerTest.php [hunk 41]
```diff
@@ -0,0 +1,121 @@
...
+            'member_id' => $create['member_id'],
+            'project_id' => $create['project_id']
+        ];
+        $response = $this->json('POST', 'memberproject/store', $array);
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+            'Failed' => 'Already exists in the database',
+            ]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberProjectControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberProjectControllerTest.php [hunk 42]
```diff
@@ -0,0 +1,121 @@
...
+    {
+        $array = [
+            'member_id' => 1,
+            'project_id' => 2
+        ];
+        $response = $this->json('POST', 'memberproject/store', $array);
+        $response
+            ->assertStatus(200)
+            ->assertJson($array);
+        $this->assertEquals(200, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberProjectControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 43]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_gender'=> [
+                        'The member gender field is required.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 44]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_gender'=> [
+                        'The selected member gender is invalid.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 45]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_position'=> [
+                        'The member position field is required.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 46]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_position'=> [
+                        'The selected member position is invalid.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 47]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_dateofbirth'=> [
+                        'The member dateofbirth must be a date after 60 years ago.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 48]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_dateofbirth'=> [
+                        'The member dateofbirth must be a date before yesterday.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 49]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_dateofbirth'=> [
+                        'The member dateofbirth is not a valid date.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 50]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_dateofbirth'=> [
+                        'The member dateofbirth field is required.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 51]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_phone'=> [
+                        'The member phone format is invalid.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 52]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_phone'=> [
+                        'The member phone may not be greater than 20 characters.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 53]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_phone'=> [
+                        'The member phone field is required.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 54]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_info'=> [
+                        'The member info may not be greater than 300 characters.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 55]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_name'=> [
+                        'The member name format is invalid.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 56]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_name'=> [
+                        'The member name may not be greater than 50 characters.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 57]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_name'=> [
+                        'The member name field is required.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 58]
```diff
@@ -0,0 +1,940 @@
...
+            'member_phone' => $array['member_phone'],
+            'member_dateofbirth' => $array['member_dateofbirth'],
+            'member_position' => $array['member_position'],
+            'member_gender' => $array['member_gender']
+        ];
+        $response = $this->json('PUT', 'member/update', $editMember);
+        $response
+            ->assertStatus(200)
+            ->assertJson($editMember);
+        $this->assertEquals(200, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 59]
```diff
@@ -0,0 +1,940 @@
...
+    {
+        $array = Factory(Member::class)->create()->toArray();
+        $this->call('DELETE', 'member/destroy', $array);
+        $response = $this->call('DELETE', 'member/destroy', $array);
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+            'Failed' => 'ID does not exist in the database',
+            ]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 60]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_gender'=> [
+                        'The selected member gender is invalid.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 61]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_gender'=> [
+                        'The member gender field is required.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 62]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_position'=> [
+                        'The selected member position is invalid.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 63]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_dateofbirth'=> [
+                        'The member dateofbirth must be a date after 60 years ago.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 64]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_dateofbirth'=> [
+                        'The member dateofbirth must be a date before yesterday.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 65]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_dateofbirth'=> [
+                        'The member dateofbirth is not a valid date.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 66]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_dateofbirth'=> [
+                        'The member dateofbirth field is required.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 67]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_phone'=> [
+                        'The member phone format is invalid.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/MemberControllerTest.php [hunk 68]
```diff
@@ -0,0 +1,940 @@
...
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+                'message'=> 'The given data was invalid.',
+                'errors' => [
+                    'member_phone'=> [
+                        'The member phone may not be greater than 20 characters.'
+                    ]
+                ]]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: This assertion has become redundant (checks the same things as above)

File: tests/Feature/ProjectControllerTest.php [hunk 69]
```diff
@@ -0,0 +1,584 @@
...
+    {
+        $array = Factory(Project::class)->create()->toArray();
+        $this->call('DELETE', 'project/destroy', $array);
+        $response = $this->call('DELETE', 'project/destroy', $array);
+        $response
+            ->assertStatus(422)
+            ->assertJson([
+            'Failed' => 'ID does not exist in the database',
+            ]);
+        $this->assertEquals(422, $response->status());
```
Comment (4c0n) [inline tests/Feature/ProjectControllerTest.php]: This assertion has become redundant (checks the same things as above)

Please check all test cases.

**Comment by tuanhai2712** [comment]
@4c0n i fix test cases as request, please check my PR, thank you so much

**Comment by tuanhai2712** [comment]
@4c0n i fixed code style errors as request, please check my PR, thank you

**Pull request merged**